core/capabilities/targets/write_target.go

```diff
@@ -0,0 +1,232 @@
+	// TODO: do spec parsing as part of parseConfig()
+
+	// Based on https://github.com/ethereum/go-ethereum/blob/f1c27c286ea2d0e110a507e5749e92d0a6144f08/signer/fourbyte/abi.go#L77-L102
+
+	// NOTE: without having full ABI it's actually impossible to support function overloading
+	selector, err := abiutil.ParseSignature(rawSelector)
+	if err != nil {
+		return nil, err
+	}
+
+	abidata, err := json.Marshal([]abi.SelectorMarshaling{selector})
+	if err != nil {
+		return nil, err
+	}
+
+	spec, err := abi.JSON(strings.NewReader(string(abidata)))
+	if err != nil {
+		return nil, err
+	}
+
+	return spec.Pack(selector.Name, args...)
+
+	// NOTE: could avoid JSON encoding/decoding the selector
+	// var args abi.Arguments
+	// for _, arg := range selector.Inputs {
+	// 	ty, err := abi.NewType(arg.Type, arg.InternalType, arg.Components)
+	// 	if err != nil {
+	// 		return nil, err
+	// 	}
+	// 	args = append(args, abi.Argument{Name: arg.Name, Type: ty})
+	// }
+	// // we only care about the name + inputs so we can compute the method ID
+	// method := abi.NewMethod(selector.Name, selector.Name, abi.Function, "nonpayable", false, false, args, nil)
+	//
+	// https://github.com/ethereum/go-ethereum/blob/f1c27c286ea2d0e110a507e5749e92d0a6144f08/accounts/abi/abi.go#L77-L82
+	// arguments, err := method.Inputs.Pack(args...)
+	// if err != nil {
+	// 	return nil, err
+	// }
+	// // Pack up the method ID too if not a constructor and return
+	// return append(method.ID, arguments...), nil
+}
+
+func (cap *EvmWrite) Execute(ctx context.Context, callback chan<- capabilities.CapabilityResponse, request capabilities.CapabilityRequest) error {
+	// TODO: technically we need to dispatch to the correct chain via network+chainID, so cap needs to store chainsets rather than a single chain
+
+	// TODO: idempotency
+
+	// TODO: extract into ChainWriter
+	txm := cap.chain.TxManager()
+
+	config := cap.chain.Config().EVM().ChainWriter()
+
+	reqConfig, err := parseConfig(request.Config)
+	if err != nil {
+		return err
+	}
+
+	inputsAny, err := request.Inputs.Unwrap()
+	if err != nil {
+		return err
+	}
+	inputs := inputsAny.(map[string]any)
+
+	// evaluate any variables in reqConfig.Params
+	args, err := evaluateParams(reqConfig.Params, inputs)
+	if err != nil {
+		return err
+	}
+
+	data, err := encodePayload(args, reqConfig.ABI)
+	if err != nil {
+		return err
+	}
+
+	gasLimit := 0
+
+	// TODO: validate encoded report is prefixed with workflowID and executionID that match the request meta
+
+	signatures := [][]byte{} //T:D:
+
+	// construct forwarding payload
+	calldata, err := forwardABI.Pack("report", common.HexToAddress(reqConfig.Address), data, signatures)
+	if err != nil {
+		return err
+	}
+	common.HexToAddress(reqConfig.Address)
+	// TODO: could we reuse forwarders code in txm (ForwardersEnabled)? it expects a different signature with no workflow meta
+
+	txMeta := &txmgr.TxMeta{
+		// FwdrDestAddress: ,
+		WorkflowExecutionID: &request.Metadata.WorkflowExecutionID,
```

Comment:
Why do we need this? Is TXM going to do anything special with the ID?

Reply:
the tx has a `SignalCallback` field that's currently used by the broadcaster to call into the task pipeline to resume a pipeline when a tx is confirmed. I'm planning to adjust that logic so that if there's a execution ID set, it'll instead call into the capability which will then send a result down the channel